{
  try {
synchronized (this) {
      if (failure == null && !closed)       wait();
      return failure == null && !closed;
    }
  }
 catch (  InterruptedException x) {
    return false;
  }
}
